### PR TITLE
Flake8-annotations: fix errors in databases

### LINKED ISSUE #310: Add flake8 plugins
- [ ] flake8 plugin to check that all function parameters have annotation
  - [ ] components: #377
  - [x] cryptography #388 
  - [x] databases #380 
  - [ ] federated
    - [ ] data #390 
    - [ ] plan #391
    - [ ] task #393 
  - [ ] interface
  - [ ] native
  - [ ] pipelines
  - [ ] plugins
  - [ ] protocols
  - [ ] transport
  - [ ] utilities
  - [ ] openfl-tutorials (optional) #383
  - [x] openfl-workspace (optional) #381 
  - [ ] final check
- [x] flake8 plugin to check that all files have correct copyrights: #370 
- [ ] flake8 plugin to check trailing comma [flake8-commas](https://pypi.org/project/flake8-commas/)

### PR DESCRIPTION
**Pull request by ViktoriiaRomanova**
Preparations to add [Flake8-annotations](https://pypi.org/project/flake8-annotations/)  plugin.

This fix is only for errors in **databases** folder.

It is assumed that the next flake8-annotations errors will be ignored:

ANN002 - Missing type annotation for `*args`
ANN003 - Missing type annotation for `**kwargs`
ANN101 - Missing type annotation for `self` in method
ANN102 - Missing type annotation for `cls` in classmethod

### CODE REVIEW

File: openfl/databases/tensor_db.py [hunk 1]
```diff
@@ -48,7 +51,7 @@ def clean_up(self, remove_older_than=1):
             self.tensor_db['round'] > current_round - remove_older_than
         ].reset_index(drop=True)
 
-    def cache_tensor(self, tensor_key_dict):
+    def cache_tensor(self, tensor_key_dict: dict) -> None:
```
Comment (dmitryagapov) [inline openfl/databases/tensor_db.py]: Dict[TensorKey, np.ndarray]

**Comment by ViktoriiaRomanova** [comment]
Lenkins please retry a build

File: openfl/databases/tensor_db.py [hunk 2]
```diff
@@ -99,8 +102,8 @@ def get_tensor_from_cache(self, tensor_key):
             return None
         return np.array(df['nparray'].iloc[0])
 
-    def get_aggregated_tensor(self, tensor_key, collaborator_weight_dict,
-                              aggregation_function):
+    def get_aggregated_tensor(self, tensor_key: TensorKey, collaborator_weight_dict: dict,
```
Comment (dmitryagapov) [inline openfl/databases/tensor_db.py:106]: `Any` -> `Type[AggregationFunction]`

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

File: openfl/databases/tensor_db.py [hunk 3]
```diff
@@ -172,7 +175,7 @@ def get_aggregated_tensor(self, tensor_key, collaborator_weight_dict,
 
         return np.array(agg_nparray)
 
-    def _iterate(self, order_by='round', ascending=False):
+    def _iterate(self, order_by: str = 'round', ascending: bool = False) -> Iterable:
```
Comment (dmitryagapov) [inline openfl/databases/tensor_db.py]: Iterator[pd.Series]

**Review by dmitryagapov — CHANGES_REQUESTED** [review]

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

File: openfl/databases/tensor_db.py (hunk 1 above)
Comment (igor-davidyuk) [reply openfl/databases/tensor_db.py]: We have a strange function here called `cache_tensor` but accepting a dict with tensors 😁

**Review by dmitryagapov — APPROVED** [review]

File: openfl/databases/tensor_db.py (hunk 2 above)
Comment (alexey-khorkin) [reply openfl/databases/tensor_db.py:106]: Why not just 
```python3
aggregation_function: AggregationFunction
``` 
?

Comment (dmitryagapov) [reply openfl/databases/tensor_db.py:106]: Could be changed to `AggregationFunction`

**Review by alexey-khorkin — APPROVED** [review]

**Review by igor-davidyuk — APPROVED** [review]

**Review by alexey-gruzdev — APPROVED** [review]

**Pull request merged**